Ball of Duty Server/Domain/Broker.cs

```diff
@@ -117,11 +207,19 @@ public void Receive()
                     }
                 });
             }
+            catch (Exception e)
```

Comment:
No, there is also a possible end of stream exception, it just isnt being handled.